insights/core/remote_resource.py

```diff
@@ -0,0 +1,142 @@
+import requests
+
+import redis
+import calendar
+from cachecontrol.heuristics import BaseHeuristic
+from cachecontrol.wrapper import CacheControl
+from cachecontrol.caches.file_cache import FileCache
+
+from datetime import datetime, timedelta
+from email.utils import parsedate, formatdate
+from cachecontrol.caches.redis_cache import RedisCache
+
+
+class RemoteResource(object):
+    """
+    RemoteResource class for accessing external Web resources.
+
+    Attributes:
+        timeout(float): Time in seconds for the requests.get api call to wait before returning a timeout exception
+
+    Examples:
+        >>> from insights.core.remote_resource import RemoteResource
+        >>> rr = RemoteResource()
+        >>> rtn = rr.get("http://google.com")
+        >>> print (rtn.content)
+    """
+
+    timeout = 10
+
+    def __init__(self, session=None):
+
+        self.session = session or requests.Session()
+
+    def get(self, url, params={}, headers={}, auth=(), verify=False):
+        """
+        Returns the response payload from the request to the given URL.
+
+        Args:
+            url (str): The URL for the WEB API that the request is being made too.
+            params (dict): Dictionary containing the query string parameters.
+            headers (dict): HTTP Headers that may be needed forthe request.
+            auth (tuple): User ID and password for Basic Auth
+            verify (str/bool): Value must be path to the Cert Bundle if verifying the SSL certificate
+             or boolean False to ignore verifying the SSL certificate.
+
+        Returns:
+            Response:(Response): Response object from requests.get api request
```

Comment:
Is the returned value a `requests` response object?  For some reason in the docs this is linking to an internal Response object when it renders.  This applies to all of the Response returns.  Normally this would be in the format `type: description` for return values.

Reply:
Corrected